backend/src/schema/resolvers/reports.js

```diff
@@ -87,6 +81,60 @@ export default {
           break
       }
 
+      return response
+    },
+  },
+  Query: {
+    reports: async (_parent, _params, { driver, _req, _user }, _resolveInfo) => {
+      const session = driver.session()
+      const res = await session.run(
+        `
+          MATCH (submitter:User)-[report:REPORTED]->(resource)
+          WHERE resource:User OR resource:Comment OR resource:Post
+          RETURN report, submitter, resource, labels(resource)[0] as type
+        `,
+        {},
+      )
+      session.close()
+
+      const dbResponse = res.records.map(r => {
+        return {
+          report: r.get('report'),
+          submitter: r.get('submitter'),
+          resource: r.get('resource'),
+          type: r.get('type'),
+        }
+      })
+      if (!dbResponse) return null
+
+      const response = []
+      dbResponse.forEach(ele => {
+        const { report, submitter, resource, type } = ele
+
+        const responseEle = {
+          ...report.properties,
+          resource: resource.properties,
+          resourceId: resource.properties.id,
```

Comment:
why do we need `resource` and also `resourceId` if the resource has the id already?

Reply:
why do we need `resource` at all actually, aren't we checking the type and returning the resources' properties based on which type it is?

Reply:
I forgot to delete the code here after I find out it can not be supported now … Thanks for the hint @mattwr18 
Done …